bundles/org.openhab.binding.modbus.sunspec/src/main/java/org/openhab/binding/modbus/sunspec/internal/handler/AbstractSunSpecHandler.java

```diff
@@ -0,0 +1,493 @@
+        if (endpoint.isPresent()) {
+            return;
+        }
+
+        ModbusEndpointThingHandler slaveEndpointThingHandler = getEndpointThingHandler();
+        if (slaveEndpointThingHandler == null) {
+            String label = Optional.ofNullable(getBridge()).map(b -> b.getLabel()).orElse("<null>");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                    String.format("Bridge '%s' is offline", label));
+            logger.debug("No bridge handler available -- aborting init for {}", label);
+            return;
+        }
+
+        try {
+            slaveId = slaveEndpointThingHandler.getSlaveId();
+
+            @Nullable
+            ModbusSlaveEndpoint optionalEndpoint = slaveEndpointThingHandler.asSlaveEndpoint();
+
+            if (optionalEndpoint == null) {
+                endpoint = Optional.empty();
+            } else {
+                endpoint = Optional.of(optionalEndpoint);
+            }
+        } catch (EndpointNotInitializedException e) {
+            // this will be handled below as endpoint remains null
+        }
+
+        if (!endpoint.isPresent()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, String.format(
+                    "Bridge '%s' not completely initialized", Optional.ofNullable(getBridge()).map(b -> b.getLabel())));
+            logger.debug("Bridge not initialized fully (no endpoint) -- aborting init for {}", this);
+            return;
+        }
+    }
+
+    /**
+     * Remove the endpoint if exists
+     */
+    private void unregisterEndpoint() {
+        if (endpoint.isPresent()) {
+            endpoint = Optional.empty();
+        }
+    }
+
+    /**
+     * Register poll task
+     * This is where we set up our regular poller
+     */
+    private synchronized void registerPollTask(ModelBlock mainBlock) {
+        if (pollTask.isPresent()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+            throw new IllegalStateException("pollTask should be unregistered before registering a new one!");
+        }
+        if (!config.isPresent() || !endpoint.isPresent()) {
+            throw new IllegalStateException("registerPollTask called without proper configuration");
+        }
+
+        logger.debug("Setting up regular polling");
+
+        BasicModbusReadRequestBlueprint request = new BasicModbusReadRequestBlueprint(getSlaveId(),
+                ModbusReadFunctionCode.READ_MULTIPLE_REGISTERS, mainBlock.address, mainBlock.length,
+                config.get().getMaxTries());
+
+        pollTask = Optional.of(new BasicPollTaskImpl(endpoint.get(), request, new ModbusReadCallback() {
+
+            @Override
+            public void onRegisters(@Nullable ModbusReadRequestBlueprint request,
+                    @Nullable ModbusRegisterArray registers) {
+                if (registers == null) {
+                    logger.info("Received empty register array on poll");
```

Comment:
does this need to be logged?

Reply:
I would like to keep it. In case there is some error in the communication with the remote device, this could be helpful in debugging the setup.

Reply:
Agreed. But please reduce to DEBUG